src/mint/v2/flexible.ts

```diff
@@ -0,0 +1,162 @@
+import algosdk from "algosdk";
+import AlgodClient from "algosdk/dist/types/src/client/v2/algod/algod";
+
+import {MINT_V2_APP_ARGUMENT, MINT_FLEXIBLE_MODE_APP_ARGUMENT} from "../ constants";
+import {CONTRACT_VERSION} from "../../contract/constants";
+import {
+  InitiatorSigner,
+  SignerTransaction,
+  SupportedNetwork
+} from "../../util/commonTypes";
+import TinymanError from "../../util/error/TinymanError";
+import {PoolInfo} from "../../util/pool/poolTypes";
+import {
+  getTxnGroupID,
+  isAlgo,
+  sendAndWaitRawTransaction,
+  sumUpTxnFees
+} from "../../util/util";
+import {getValidatorAppID} from "../../validator";
+import {MintExecution, MintTxnIndices} from "../types";
+
+export async function generateTxns({
+  client,
+  pool,
+  network,
+  poolAddress,
+  asset_1,
+  asset_2,
+  liquidityToken,
+  initiatorAddr
+}: {
+  client: AlgodClient;
+  pool: PoolInfo;
+  network: SupportedNetwork;
+  poolAddress: string;
+  asset_1: {id: number; amount: number | bigint};
+  asset_2: {id: number; amount: number | bigint};
+  liquidityToken: {id: number; amount: number | bigint};
+  slippage: number;
+  initiatorAddr: string;
+}) {
+  const suggestedParams = await client.getTransactionParams().do();
+  const isAlgoPool = isAlgo(asset_2.id);
+  const asset1InTxn = algosdk.makeAssetTransferTxnWithSuggestedParamsFromObject({
+    from: initiatorAddr,
+    to: poolAddress,
+    assetIndex: pool.asset1ID,
+    amount: asset_1.amount,
+    suggestedParams
+  });
+  const asset2InTxn = isAlgoPool
+    ? algosdk.makePaymentTxnWithSuggestedParamsFromObject({
+        from: initiatorAddr,
+        to: poolAddress,
+        amount: asset_2.amount,
+        suggestedParams
+      })
+    : algosdk.makeAssetTransferTxnWithSuggestedParamsFromObject({
+        from: initiatorAddr,
+        to: poolAddress,
+        assetIndex: pool.asset2ID,
+        amount: asset_2.amount,
+        suggestedParams
+      });
+  const validatorAppCallTxn = algosdk.makeApplicationNoOpTxnFromObject({
+    from: poolAddress,
+    appIndex: getValidatorAppID(network, CONTRACT_VERSION.V2),
+    appArgs: [MINT_V2_APP_ARGUMENT, MINT_FLEXIBLE_MODE_APP_ARGUMENT],
+    accounts: [poolAddress],
+    foreignAssets: [liquidityToken.id],
+    suggestedParams: {
+      ...suggestedParams,
+      // In addition to the AppCall txn, there will be two additional Inner Transactions.
+      fee: 3 * suggestedParams.fee
```

Comment:
```suggestion
      // Add +1 to account for the fee of the outer txn
      fee: (V2_MINT_INNER_TXN_COUNT + 1) * ALGORAND_MIN_TX_FEE
```
- I used this approach in my prs, i think it's more easy to understand what's going on here
- As I mentioned, looks like `suggestedParams.fee` is not doing what we want, `ALGORAND_MIN_TX_FEE` we can use ALGORAND_MIN_TX_FEE